addons/web/static/src/js/views/kanban/kanban_record.js

```diff
@@ -566,14 +568,25 @@ var KanbanRecord = Widget.extend({
                 break;
             case 'action':
             case 'object':
-                this.trigger_up('button_clicked', {
-                    attrs: $action.data(),
-                    record: this.state,
-                });
+                var confirm = $(event.currentTarget).attr("confirm");
+                if (confirm) {
+                    Dialog.confirm(this, confirm, {
+                        confirm_callback: object_trigger_up
+                    });
+                }
+                else {
+                    object_trigger_up()
+                }
                 break;
             default:
                 this.do_warn("Kanban: no action for type : " + type);
         }
+        function object_trigger_up () {
+            _this.trigger_up('button_clicked', {
```

Comment:
Who is `_this`?

Reply:
Ah.  Now I see that `_this` is declared a few lines above.  I would reorganize the code to make this evident.  I would put the definition of `object_trigger_up` right below a  `var self = this;` sentence.  The change to `self` is just to be a nice neighbor.